camera_calibration/src/camera_calibration/camera_calibrator.py

```diff
@@ -229,8 +230,15 @@ def handle_monocular(self, msg):
                                         checkerboard_flags=self.checkerboard_flags,
                                         max_chessboard_speed = self._max_chessboard_speed)
 
-        # This should just call the MonoCalibrator
-        drawable = self.c.handle_msg(msg)
+        # This should just call the MonoCalibrator and compute the mask for the detected corners if any
+        drawable = self.c.handle_msg(msg, self._mask)
+        if self.c.mask is not None and self._mask:
+            # Transforming the mask in a BGR image
+            blue_layer = numpy.zeros(self.c.mask.shape, dtype=numpy.uint8)
```

Comment:
instead of creating 4 different arrays you can just initialize the bgr image with zeros and update the red and green layer with mask * 255 and ~mask*255